src/cli/getInformationFromIndex.js

```diff
@@ -0,0 +1,40 @@
+const algoliasearch = require('algoliasearch');
+const { createInMemoryCache } = require('@algolia/cache-in-memory');
+
+const clients = new Map();
+function getClient(appId, apiKey) {
+  const key = [appId, apiKey].join('__');
+  let client = clients.get(key);
+  if (!client) {
+    client = algoliasearch(appId, apiKey, {
+      responsesCache: createInMemoryCache(),
+      requestsCache: createInMemoryCache(),
+    });
+
+    clients.set(key, client);
+  }
+
+  return client;
+}
+
+async function getInformationFromIndex({ appId, apiKey, indexName }) {
```

Comment:
What about passing the initialized search client instead? This way you wouldn't have to use a "manual" singleton.

```diff
- async function getInformationFromIndex({ appId, apiKey, indexName }) {
+ async function getInformationFromIndex({ searchClient, indexName }) {
```

Reply:
there's no place in the control flow to save a variable like that, all answers from other questions are forwarded, but you'd need something like a local mutable variable in the "answer questions" flow to save the search client

Reply:
Yeah indeed I just saw that there's no callback at the question-level in the API, or any kind of context 😞